### ISSUE TITLE
Allow arrowheads in the middle of a shaft

### ISSUE THREAD

**Issue by byorgey** [diagrams/diagrams-lib#262]
Sometimes one might like to draw arrows where the arrowhead shows up at the *midpoint* of the shaft.  For example, this style is often used to indicate the orientation of edges in a directed graph.  Currently this can be hacked by e.g. drawing two arrows, one to the midpoint and one without a head the rest of the way (overlapping them a bit so there is no gap) but it would be nice to support this directly.  I am not sure what the API should be.  Perhaps there can just be a parameter specifying how far along the shaft the head (or, respectively, the tail) should be placed (as a fraction of the arc length); if you want the midpoint you can specify 0.5.

**Comment by jeffreyrosenbluth** [comment]
More generally most vector graphics libraries allow markers to be placed
anywhere along a "path".
I believe most packages place the marker on top of the path, in contrast to
our arrow implementation,
in which the shaft is carefully connected to the arrowhead.

**Comment by byorgey** [comment]
Right, so maybe the more general issue is that we need a nicer way to draw
arrowheads/tails/other markers independently of an arrow.
